### PR TITLE
[FIX] business_requirement_deliverable_cost

### PR DESCRIPTION
**Pull request by martintorresvm**
- Fix coversion of UoM
- Fix on_change functions
- Centralize pricelist based product price calculation
- Total revenue from Delirable should be replace by Resources when this module is installed

### CODE REVIEW

File: business_requirement_deliverable_cost/models/business.py [hunk 1]
```diff
@@ -41,6 +41,35 @@ class BusinessRequirementResource(models.Model):
...
+                    lang=resource.partner_id.lang,
+                    partner=resource.partner_id.id,
+                    quantity=resource.qty,
+                    pricelist=pricelist_id.id,
+                    uom=resource.uom_id.id,
+                )
+                resource.sale_price_unit = product.price
+
+    @api.multi
+    def _set_unit_price(self):
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: _set_cost_price

File: business_requirement_deliverable_cost/models/business.py [hunk 2]
```diff
@@ -257,5 +253,7 @@ def _compute_resource_procurement_total(self):
         'resource_procurement_total')
     def _compute_gross_profit(self):
         for br in self:
-            br.gross_profit = br.total_revenue - \
-                br.resource_task_total - br.resource_procurement_total
+            br.gross_profit = (
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py:247]: better?
```
br.gross_profit =
                br.total_revenue
                - br.resource_task_total
                - br.resource_procurement_total
```

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details but overall LGTM
@seb-elico

File: business_requirement_deliverable_cost/models/business.py [hunk 3]
```diff
@@ -180,18 +176,17 @@ def action_button_update_estimation(self):
...
-                        resource.sale_price_unit = product.price
+                    resource._set_sales_price()
+                    resource._set_unit_price()
+
+    @api.multi
+    @api.depends('sale_price_unit', 'qty', 'resource_ids')
+    def _compute_get_price_total(self):
+        # Re Compute the BRD Total Revenue with RL sales price
+        super(BusinessRequirementDeliverable, self)._compute_get_price_total()
+        for brd in self:
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: I am not sure this is desired. to be discussed.

File: business_requirement_deliverable_cost/models/business.py (hunk 2 above)
Comment (seb-elico) [reply business_requirement_deliverable_cost/models/business.py:247]: Yes it's easier to read like this IMO (though you still need the parenthesis), unfortunately PEP8 doesn't force this syntax

**Review by seb-elico — COMMENTED** [review]
LGTM after Eric's comments are addressed

File: business_requirement_deliverable_cost/models/business.py (hunk 2 above)
Comment (martintorresvm) [reply business_requirement_deliverable_cost/models/business.py:247]: More human readable IMO. Better to understand

**Comment by martintorresvm** [comment]
@seb-elico @elicoidal My fuctntional test looks ok and within the other PR #135 

Could you approve these changes?

I'll probable need fix Travis

**Review by elicoidal — APPROVED** [review]

**Comment by elicoidal** [comment]
OK: Moving forward for the bug fixing

**Pull request merged**